cdap-app-fabric/src/main/java/io/cdap/cdap/internal/credential/handler/CredentialProviderHttpHandler.java

```diff
@@ -361,15 +392,18 @@ public void deleteIdentity(HttpRequest request, HttpResponder responder,
 
   private void ensureNamespaceExists(String namespace)
       throws IOException, NamespaceNotFoundException {
+    NamespaceId namespaceId;
+    Boolean namespaceFound;
     try {
-      NamespaceId namespaceId = new NamespaceId(namespace);
```

Comment:
why was this change made?

Reply:
NamespaceNotFoundException originally got caught by the Exception handler in the try-catch and wrapped in an IOException unintentionally.